lib/faktory_worker/worker/heartbeat_server.ex

```diff
@@ -67,6 +67,12 @@ defmodule FaktoryWorker.Worker.HeartbeatServer do
     {:noreply, state, {:continue, :schedule_beat}}
   end
 
+  # Crash the server, allowing the Supervisor to restart it so we attempt to
+  # create a new connection when Faktory is disrupted.
+  def handle_info(:beat, %{beat_state: :error} = state) do
+    {:stop, :invalid_connection, state}
```

Comment:
Should there be any kind of delay or exponential backoff that happens here?  I'm not sure how quickly the heartbeat server starts back up again and attempts reconnection.  If it's thousands of times per second (very possible with Elixir), we'll probably want to slow it down.

Reply:
It's every 15 seconds - this will crash the server, which will immediately be restarted, and then the startup process for the server will immediately attempt to create a new connection and then schedule a heartbeat for 15 seconds from then. So, no exponential backoff or anything, but one attempt every 15 seconds even in cases where Faktory is actually down doesn't feel too bad to me.